src/components/templates/PartyMap/components/Map/Map.tsx

```diff
@@ -1,199 +1,181 @@
-  const [selectedUserProfile, setSelectedUserProfile] = useState<
-    WithId<User>
-  >();
-  const [rows, setRows] = useState<number>(0);
-
-  const columns = venue.columns ?? DEFAULT_COLUMNS;
-  const currentPosition = profile?.data?.[venue.id];
+  const userUid = user?.uid;
+  const showGrid = venue.showGrid;
 
-  const columnsArray = useMemo(() => Array.from(Array<JSX.Element>(columns)), [
-    columns,
-  ]);
-  const rowsArray = useMemo(() => Array.from(Array(rows)), [rows]);
+  const totalColumns = venue.columns ?? DEFAULT_COLUMNS;
+  const [totalRows, setTotalRows] = useState<number>(0);
 
-  const venues = useSelector(orderedVenuesSelector);
-  const partygoers = usePartygoers();
+  const columnsArray = useMemo(
+    () => Array.from(Array<JSX.Element>(totalColumns)),
+    [totalColumns]
+  );
+  const rowsArray = useMemo(() => Array.from(Array(totalRows)), [totalRows]);
 
   useEffect(() => {
     const img = new Image();
     img.src = venue.mapBackgroundImageUrl ?? "";
     img.onload = () => {
       const imgRatio = img.width ? img.width / img.height : 1;
+
       const calcRows = venue.columns
         ? Math.round(parseInt(venue.columns.toString()) / imgRatio)
         : DEFAULT_ROWS;
-      setRows(calcRows);
+
+      setTotalRows(calcRows);
     };
   }, [venue.columns, venue.mapBackgroundImageUrl]);
 
-  const roomsHit = useMemo(() => {
-    if (!currentPosition?.row || !currentPosition?.column) return [];
-
-    const { row, column } = currentPosition;
-
-    const roomHitFilter = makeCampRoomHitFilter({
-      row,
-      column,
-      totalRows: rows,
-      totalColumns: columns,
-    });
-
-    return venue.rooms?.filter(roomHitFilter);
-  }, [currentPosition, rows, columns, venue.rooms]);
-
-  const detectRoomsOnMove = useCallback(() => {
-    if (selectedRoom) {
-      const noRoomHits = hasElements(roomsHit);
-      if (!noRoomHits && selectedRoom) {
-        setSelectedRoom(undefined);
-        setIsRoomModalOpen(false);
-      }
-    }
-
-    roomsHit?.forEach((room) => {
-      setSelectedRoom(room);
-      setIsRoomModalOpen(true);
-    });
-  }, [roomsHit, selectedRoom, setIsRoomModalOpen, setSelectedRoom]);
-
   const takeSeat = useCallback(
     (row: number | null, column: number | null) => {
-      if (!user || !profile || !venueId) return;
-      const doc = `users/${user.uid}`;
-      const existingData = profile?.data;
-      const update = {
-        data: {
-          ...existingData,
-          [venueId]: {
-            row,
-            column,
-          },
-        },
-      };
-      const firestore = firebase.firestore();
-      firestore
-        .doc(doc)
-        .update(update)
-        .catch(() => {
-          firestore.doc(doc).set(update);
-        });
+      makeUpdateUserGridLocation({
+        venueId,
+        userUid: user?.uid,
+        profileData,
+      })(row, column);
     },
-    [profile, user, venueId]
+    [profileData, user, venueId]
   );
```

Comment:
Can we refactor this to use `undefined` instead of `null`?

Reply:
I think for the same reason as https://github.com/sparkletown/sparkle/pull/1080#discussion_r559946760 this is probably OOS for this PR